File: ontrack-extension-workflows/src/main/java/net/nemerosa/ontrack/extension/workflows/engine/WorkflowInstanceStatus.kt
package net.nemerosa.ontrack.extension.workflows.engine

enum class WorkflowInstanceStatus(
    val finished: Boolean,
) {

    STARTED(
        finished = false,
    ),
    RUNNING(
        finished = false,
    ),
    STOPPED(
        finished = true,
    ),
    ERROR(
        finished = true,
    ),
    SUCCESS(
        finished = true,
    ),

}

File: ontrack-web-core/components/extension/workflows/WorkflowInstanceStatus.js
import {Space} from "antd";
import {FaCheck, FaSpinner, FaStop, FaTimesCircle} from "react-icons/fa";

export default function WorkflowInstanceStatus({status}) {
    return (
        <>
            {
                status === 'STARTED' &&
                <Space>
                    <FaSpinner color="orange"/>
                    Started
                </Space>
            }
            {
                status === 'RUNNING' &&
                <Space>
                    <FaSpinner color="blue"/>
                    Running
                </Space>
            }
            {
                status === 'STOPPED' &&
                <Space>
                    <FaStop color="red"/>
                    Stopped
                </Space>
            }
            {
                status === 'ERROR' &&
                <Space>
                    <FaTimesCircle color="red"/>
                    Error
                </Space>
            }
            {
                status === 'SUCCESS' &&
                <Space>
                    <FaCheck color="green"/>
                    Success
                </Space>
            }
        </>
    )
}